dot/parachain/backing/statement_table.go

```diff
@@ -380,16 +380,17 @@ func effectiveMinimumBackingVotes(groupLen uint, configuredMinimumBackingVotes u
 	return min(groupLen, uint(configuredMinimumBackingVotes))
 }
 
-func (statementTable) drainMisbehaviors() []parachaintypes.ProvisionableDataMisbehaviorReport {
-	// TODO: Implement this method
-	return nil
+func (table *statementTable) drainMisbehaviors() map[parachaintypes.ValidatorIndex][]parachaintypes.Misbehaviour {
```

Comment:
Perhaps you could add a comment for this function, even though it's un-exported and a basic function, it's nice to have comments for these.